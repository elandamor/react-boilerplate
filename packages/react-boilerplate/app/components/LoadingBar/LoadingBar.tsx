import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

interface IProps {
  className?: string;
}

const LoadingBar: SFC<IProps> = ({ className }) => (
  <Wrapper className={classNames('c-loadingBar', className)} />
);

LoadingBar.defaultProps = {
  className: '',
};

export default LoadingBar;
