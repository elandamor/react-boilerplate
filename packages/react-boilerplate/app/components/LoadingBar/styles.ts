import styled from 'styled-components';

const Wrapper = styled.div`
  background: var(--brand-success);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transform: translateX(100%);
  z-index: 10;
`;

export default Wrapper;
