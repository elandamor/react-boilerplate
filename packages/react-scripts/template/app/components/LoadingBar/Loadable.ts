/**
 *
 * Asynchronously loads the component for LoadingBar
 *
 */

import React from 'react';

// @ts-ignore
export default React.lazy(() => import('./index'));
