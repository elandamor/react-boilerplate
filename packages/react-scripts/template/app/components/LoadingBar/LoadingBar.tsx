import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

interface IProps {
  className?: string;
  loading?: boolean;
}

const LoadingBar: SFC<IProps> = ({ className, loading }) => (
  <Wrapper
    className={classNames('c-loadingBar', className, {
      '-loading': loading,
    })}
  />
);

export default LoadingBar;
