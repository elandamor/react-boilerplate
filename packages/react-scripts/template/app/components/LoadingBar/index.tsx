import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

export interface IProps {
  className?: string;
  loading?: boolean;
}

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar loading />
 */

const LoadingBar: SFC<IProps> = ({ className, loading }) => (
  <Wrapper
    className={classNames('c-loadingBar', className)}
    {...( loading ? { 'data-loading': true } : { 'data-loading': false } )}
  />
);

export default LoadingBar;
