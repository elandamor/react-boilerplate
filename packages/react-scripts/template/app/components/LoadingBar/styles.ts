import styled, { css, keyframes } from 'styled-components';
import { IProps } from './index';

const shiftRightwards = keyframes`
  0% { transform: translateX(-100%) }
  40% { transform: translateX(0) }
  60% { transform: translateX(0) }
  100% { transform: translateX(100%) }
`;

const Wrapper = styled.div`
  background: var(--brand-success, #000000);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transform: translateX(100%);
  z-index: 10;

  ${(props: IProps) =>
    props.loading &&
    css`
      animation: ${shiftRightwards} 1s ease-in-out infinite;
      animation-delay: 0.8s;
      display: block;
    `};
`;

export default Wrapper;
