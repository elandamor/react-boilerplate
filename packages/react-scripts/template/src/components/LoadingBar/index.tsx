import styled, { keyframes } from 'styled-components';

const shiftRightwards = keyframes`
  0% { transform: translateX(-100%) }
  40% { transform: translateX(0) }
  60% { transform: translateX(0) }
  100% { transform: translateX(100%) }
`;

/**
 * @render react
 * @name LoadingBar component
 * @description LoadingBar component.
 * @example
 * <LoadingBar />
 */

const LoadingBar = styled.div`
  animation: ${shiftRightwards} 1s ease-in-out infinite;
  animation-delay: 0.8s;
  background: var(--brand-success, #000000);
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  transform: translateX(100%);
  z-index: 10;
`;

export default LoadingBar;
