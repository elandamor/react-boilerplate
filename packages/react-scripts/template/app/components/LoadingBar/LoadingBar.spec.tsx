// LoadingBar.spec.tsx
import * as React from 'react';
import { cleanup, render } from 'react-testing-library';

import LoadingBar from './index';

afterEach(cleanup);

describe('LoadingBar', () => {
  it('should render loading bar (not loading)', () => {
    // TODO: Check for <data-loading=false>
    render(<LoadingBar />);
  });

  it('should render loading bar (loading)', () => {
    // TODO: Check for <data-loading=true>
    render(<LoadingBar loading />);
  });
});
