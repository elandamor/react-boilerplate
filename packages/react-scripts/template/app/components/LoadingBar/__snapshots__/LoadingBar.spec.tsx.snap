// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingBar /> should display a loading bar 1`] = `
.c0 {
  background: var(--brand-success);
  display: none;
  height: 2px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  z-index: 10;
}

<div
  className="c-loadingBar c0"
/>
`;
